Polyhedron/demo/Polyhedron/Viewer.cpp

```diff
@@ -1365,7 +1536,112 @@ void Viewer::messageLogged(QOpenGLDebugMessage msg)
   qDebug() << qPrintable(error) << "\n" << qPrintable(msg.message()) << "\n";
 }
 
+void Viewer::setLighting()
+{
+  
+  //save current settings;
+  float prev_spec = d->spec_power;
+  QVector4D prev_pos = d->position;
+  QVector4D prev_ambient = d->ambient;
+  QVector4D prev_diffuse = d->diffuse;
+  QVector4D prev_spec_color = d->specular;
+  //open dialog
+  LightingDialog* dialog = new LightingDialog(d);
+  //set specular
+  connect(dialog->spec_powrSlider, &QSlider::valueChanged,
+          [this, dialog]()
+  {
+    d->spec_power = dialog->spec_powrSlider->value();
+    update();
+  });
+  //set position
+  connect(dialog->position_lineEdit, &QLineEdit::editingFinished,
+          [this, dialog]()
+  {
+    QStringList list = dialog->position_lineEdit->text().split(QRegExp(","), QString::SkipEmptyParts);
+    if (list.isEmpty()) return;
+    if (list.size()!=3){
+      QMessageBox *msgBox = new QMessageBox;
+      msgBox->setWindowTitle("Error");
+      msgBox->setText("ERROR : Input should consists of 3 floats.");
+      msgBox->exec();
+      return;
+    }
+    double coords[3];
+    for(int j=0; j<3; ++j)
+    {
+      bool ok;
+      coords[j] = list.at(j).toFloat(&ok);
+      if(!ok)
+      {
+          QMessageBox *msgBox = new QMessageBox;
+          msgBox->setWindowTitle("Error");
+          msgBox->setText("ERROR : Coordinates are invalid.");
+          msgBox->exec();
+          return;
+      }
+    }
+    d->position = QVector4D(coords[0], coords[1], coords[2], 1.0f);
+    update();
+  });
+  
+
+  //set ambient
+  connect(dialog, &LightingDialog::s_ambient_changed,
+          [this, dialog](){
+    d->ambient=QVector4D(dialog->ambient.redF(), 
+                         dialog->ambient.greenF(), 
+                         dialog->ambient.blueF(), 
+                         1.0f);
+    update();
+  });
+          
+  //set diffuse
+  connect(dialog, &LightingDialog::s_diffuse_changed,
+          [this, dialog](){
+    d->diffuse=QVector4D(dialog->diffuse.redF(), 
+                         dialog->diffuse.greenF(), 
+                         dialog->diffuse.blueF(), 
+                         1.0f);
+    update();
+  });
+  //set specular
+  connect(dialog, &LightingDialog::s_specular_changed,
+          [this, dialog](){
+    d->specular=QVector4D(dialog->specular.redF() , 
+                         dialog->specular.greenF(), 
+                         dialog->specular.blueF() , 
+                         1.0f);
+    update();
+    
+  });
+  
+  //reset default
+  connect(dialog->buttonBox->button(QDialogButtonBox::StandardButton::RestoreDefaults), &QPushButton::clicked,
+          [this, dialog](){
+    d->position = QVector4D(0,0,1,1);
+    d->ambient=QVector4D(77.0/255,77.0/255,77.0/255, 1.0);
+    d->diffuse=QVector4D(204.0/255,204.0/255,204.0/255,1.0);
+    d->specular=QVector4D(0,0,0,1.0);
+    d->spec_power = 51;    
+    update();
+    
+  });
```

Comment:
There is [a warning](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13-Ic-75/Polyhedron_Demo/TestReport_lrineau_ArchLinux-clang-CXX14.gz):
```
.../ArchLinux-clang-CXX14/test/Polyhedron_Demo/Viewer.cpp:1621:18: warning: lambda capture 'dialog' is not used [-Wunused-lambda-capture]
          [this, dialog](){
                 ^
1 warning generated.
```